### ISSUE TITLE
-Werror=use-without-only not supported in few version of gcc

### ISSUE THREAD

**Issue by bishtgautam** [BeTR-biogeochemistry-modeling/sbetr#1]
`-Werror=use-without-only` not supported in gnu 4.9.3, but is supported in 5.1.0. Is there a minimum gcc version requirement?

**Comment by bandre-ucar** [comment]
Um.... I'm using 5.3, I think Jinyun uses 6.1? I'm working on getting travis-ci working now that we have a public repo. Whatever they support will be what we support....

**Comment by bishtgautam** [comment]
Not a big deal. I will just upgrade the compiler on my machine over the weekend to 5.x.

**Comment by bandre-ucar** [comment]
No, we need to specify build dependancies for the compiler and netcdf. I just haven't made it a priority. If nothing else, we should add some build flags based on compiler version.

**Comment by bandre-ucar** [comment]
What's the minimum version of gfortran that acme supports? My understanding is that cesm is going with something in the 5.x range.

**Comment by jinyun1tang** [comment]
The minimum version is more like some 5.x version. I previously stumbled with older versions and decided to go with 6.x versions.

**Comment by bishtgautam** [comment]
acme-se teams says the minimum supported version of gnu is 5.3. Should I close this issue or one wants to support gnu < 5.x?

**Comment by bandre-ucar** [comment]
I think the gfortran version for travis is about 4.8. I'm not sure how much
of a pain it will be to get 5.x installed. Let me get things working there
and then I'll resolve this issue.

Ben

**Comment by bandre-ucar** [comment]
I want to keep the stricter compiler warnings in gfortran >= 5 because they help catch bad coding practices. The correct way to handle this is with feature discovery. Tell cmake to test if the various compiler options are available in the compiler, and add them to the fortran flags only if they are. If we implement that, then we could support gfortran 4.9. I looked at the cmake doc for this, and I don't think it's terribly hard to implement, but I can't really justify the time right now.
